cpp/test/preprocess_utils.cu

```diff
@@ -0,0 +1,283 @@
+/*
```

Comment:
This file adds in some utilities that I needed to conduct the randomized Rmat matrix generation along with the logic for the kernel that calculates tfidf/bm25. Since I have two test files, I took out all the commonalities of the calculations and placed them here, so both files could use the same logic.

Reply:
If this is a batched deduped version of rmat, that's super useful. We don't have to do this now, but can you create an issue to move these into a first-class API and then reference the issue here so we don't lose sight of it?

Reply:
it is the elements of the batched deduped rmat. We just need to run them in a while loop (and changing the random seed) until we actually get the number of vertices that we are targeting.